tools/booster/Proxy.hs

```diff
@@ -295,8 +343,13 @@ respondEither mbStatsVar booster kore req = case req of
                         | null filteredNexts ->
                             -- HACK. Would want to return the prior state
                             res{reason = Stuck, nextStates = Nothing}
-                    _otherReason ->
-                        res{state = simplifiedState, nextStates}
+                    _otherReason
+                        | null filteredNexts ->
+                            res{state = simplifiedState, nextStates = Nothing}
+                        | otherwise ->
+                            res{state = simplifiedState, nextStates = Just filteredNexts}
+            -- NOTE: we deliberately ignore simplification traces produced by simplifyExecuteState.
+            --       append them to result.logs if needed in the future.
```

Comment:
:raised_eyebrow: Why that? I think the simplification logs must be appended (just like we collect them inside  `RewriteSimplified` during/after  rewrites, see [related code](https://github.com/runtimeverification/hs-backend-booster/blob/main/library/Booster/Pattern/Rewrite.hs#L580) )

Reply:
Yeah, good point. Will add them.